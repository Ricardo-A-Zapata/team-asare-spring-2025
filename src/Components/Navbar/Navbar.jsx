import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useAuth } from '../../AuthContext';

export const PAGES = [
  { label: 'Home', destination: '/' },
  { label: 'About', destination: '/about' },
  { label: 'Masthead', destination: '/masthead' },
  { label: 'View All Users', destination: '/users' },
  { label: 'View All Manuscripts', destination: '/manuscripts' },
];

function NavLink({ page }) {
  const { label, destination } = page;
  return (
    <li>
      <Link to={destination}>{label}</Link>
    </li>
  );
}
NavLink.propTypes = {
  page: propTypes.shape({
    label: propTypes.string.isRequired,
    destination: propTypes.string.isRequired,
  }).isRequired,
};

function Navbar() {
  const { isLoggedIn } = useAuth();
  
  return (
    <nav>
      <ul className="wrapper">
        {PAGES.map((page) => <NavLink key={page.destination} page={page} />)}
        <NavLink 
          page={{ 
            label: isLoggedIn ? 'Account' : 'Login', 
            destination: '/login'
          }} 
        />
        {!isLoggedIn && (
          <NavLink 
            page={{ 
              label: 'Sign up', 
              destination: '/signup'
            }} 
          />
        )}
      </ul>
    </nav>
  );
}

export default Navbar;
