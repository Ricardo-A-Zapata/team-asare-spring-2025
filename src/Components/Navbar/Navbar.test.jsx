import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';
import '@testing-library/jest-dom';
import { UNSAFE_logV6DeprecationWarnings } from 'react-router';
import { PAGES } from './Navbar';


UNSAFE_logV6DeprecationWarnings(false);


describe('Navbar Component', () => {
  test('renders without crashing', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
  });

  test('renders correct number of navigation links', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );

    const links = screen.getAllByRole('link');
    expect(links.length).toBe(PAGES.length);
  });

  test('renders navigation links with correct text and href attributes', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );

    const homeLink = screen.getByRole('link', { name: 'Home' });
    const usersLink = screen.getByRole('link', { name: 'View All Users' });
    const submissionsLink = screen.getByRole('link', { name: 'View All Submissions' });
    const manuscriptsLink = screen.getByRole('link', { name: 'View All Manuscripts' });

    expect(homeLink).toHaveAttribute('href', '/');
    expect(usersLink).toHaveAttribute('href', '/users');
    expect(submissionsLink).toHaveAttribute('href', '/submissions');
    expect(manuscriptsLink).toHaveAttribute('href', '/manuscripts');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
