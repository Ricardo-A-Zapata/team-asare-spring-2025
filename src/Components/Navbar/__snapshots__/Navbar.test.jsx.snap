// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="wrapper"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/users"
        >
          View All Users
        </a>
      </li>
      <li>
        <a
          href="/submissions"
        >
          View All Submissions
        </a>
      </li>
      <li>
        <a
          href="/manuscripts"
        >
          View All Manuscripts
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
